data/shared_state_variable.go

```diff
@@ -217,6 +248,12 @@ func (c *counterInt64) TryUpdateIfStale(ctx context.Context, staleness time.Dura
 		currentValue := c.value.Load()
 		if newValue > currentValue {
 			c.setValue(newValue)
+		} else if currentValue > newValue && (currentValue-newValue > c.ignoreDownDriftOf) {
+			c.setValue(newValue)
+
+			if c.driftCallback != nil {
+				c.driftCallback(currentValue, newValue)
+			}
```

Comment:
Looks like the logic of "if it's higher considering the drift then set value and call callback" is repeated few times in this func

Is it possible to simplify, in one place (using some vars), so it's easier to reason about?